src/prefect/tasks/docker/containers.py

```diff
@@ -299,3 +300,79 @@ def run(
         client = docker.APIClient(base_url=docker_server_url, version="auto")
 
         client.stop(container=container_id)
+
+
+class WaitOnContainer(Task):
+    """
+    Task for waiting on an already started Docker container.
+    Note that all initialization arguments can optionally be provided or overwritten at runtime.
+
+    Args:
+        - container_id (str, optional): The id of a container to start
+        - docker_server_url (str, optional): URL for the Docker server. Defaults to
+            `unix:///var/run/docker.sock` however other hosts such as `tcp://0.0.0.0:2375`
+            can be provided
+        - raise_on_exit_code (bool, optional): whether to raise a `FAIL` signal for a nonzero exit code;
+            defaults to `True`
+        - **kwargs (dict, optional): additional keyword arguments to pass to the Task
+            constructor
+    """
+
+    def __init__(
+        self,
+        container_id: str = None,
+        docker_server_url: str = "unix:///var/run/docker.sock",
+        raise_on_exit_code: bool = True,
+        **kwargs: Any
+    ):
+        self.container_id = container_id
+        self.docker_server_url = docker_server_url
+        self.raise_on_exit_code = raise_on_exit_code
+
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("container_id", "docker_server_url", "raise_on_exit_code")
+    def run(
+        self,
+        container_id: str = None,
+        docker_server_url: str = "unix:///var/run/docker.sock",
+        raise_on_exit_code: bool = True,
+    ) -> None:
+        """
+        Task run method.
+
+        Args:
+            - container_id (str, optional): The id of a container to start
+            - docker_server_url (str, optional): URL for the Docker server. Defaults to
+                `unix:///var/run/docker.sock` however other hosts such as `tcp://0.0.0.0:2375`
+                can be provided
+            - raise_on_exit_code (bool, optional): whether to raise a `FAIL` signal for a nonzero exit code;
+                defaults to `True`
+
+        Returns:
+            - dict: a dictionary with `StatusCode` and `Error` keys
```

Comment:
Will any `Error` keys be in the dict or will they trigger the nonzero exit code?

Reply:
There is still an `Error` key, but its value is `None`.  I haven't seen a situation where `Error` is not `None` and `StatusCode == 0`, I assume that can't happen unless the container is not even started or something. (in which case I wonder if I should put the `client.logs` inside a `try / except`....